### DISCUSSION TITLE
Plugin catches interference (Hairless + LoopMIDI + MIDI2LR)

### DISCUSSION THREAD

**Discussion by lexa2hk** [rsjaffe/MIDI2LR#742] [Problems]
Greetings to all! I am engaged in resuscitation of the Arduino project (https://github.com/rninov/LRduino ) on the OLED and custom libraries.

At this stage, the bundle is: Uno on the breadboard -> Hairless MIDI -> loopMIDI -> MIDI2LR. When the encoder is spinning, correct data is logged in Hairless, but the plugin catches interference (Puckup mode don't help at all) 

If there are experienced readers in this matter, I ask for help :)

Here's screen video (and a bit of Russian interface, but i hope thats not a big deal)

(I know what is better to use Pro Micro + MIDIUSB but now I am making project from what is at hand)
https://github.com/rsjaffe/MIDI2LR/assets/32361833/54c9b074-c0c2-4252-8e36-3cead7b20a8f

**Comment by remov-b4-flight** [comment]
Hi
I have similar project 
(https://github.com/remov-b4-flight/LrTMAX)
that uses STM32,
It looks related issues that I experienced last month.
When encoder moves, MIDI CC sent value 64,65,66 ... 
MIDI2LR App gets and reflect Lightroom development via plugin.
When encoder stop, App detects it and send CC message value 64(median) to virtual knob to center position.

It seems kind of status racing exists on CC send and receive.

Thanks.

**Reply by lexa2hk** [reply]
Thanks a lot! 

I tried to change midi note to 0 which has frequency `8.18 Hz` instead of `12543.85 Hz` on note 127.

Then i saw that **hairless sometimes beep green when i touch Breadboard cable**, so my problem was as hardware-based as possible :)

![image](https://github.com/rsjaffe/MIDI2LR/assets/32361833/43b335df-1049-4924-923e-fa071ede5349)

**Comment by remov-b4-flight** [comment]
Yet another possibility is insufficient de-chattering.
Commonly rotary encoders have 2~8ms chattering time.
It should be ignore port changes in the periods in port scan routine, 
or connect capacitors (approx. ~47nF) encoder signals between ground.  
or use both .

**Comment by hnaef01** [comment]
Could it be the same issue as here:
https://github.com/rsjaffe/MIDI2LR/discussions/727

Involved software is also loopmidi - maybe they don't work together correctly?
